### ISSUE TITLE
interface conversion: interface {} is *alpm.Package, not aur.Pkg

### ISSUE THREAD

**Issue by segln** [Jguer/yay#2142]
### Affected Version

yay v12.0.4 - libalpm v13.0.2

### Describe the bug

When I search 'duc', yay throws error:
```
$ yay -Ss duc
 -> interface conversion: interface {} is *alpm.Package, not aur.Pkg
```

### Reproduction Steps

1. `yay -Ss duc`

### Expected behavior

Information of `duc` package.

### Output

```sh
$ yay -Sys duc                                            2023/04/29 12:57:47
 -> interface conversion: interface {} is *alpm.Package, not aur.Pkg
goroutine 1 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
        runtime/debug/stack.go:16 +0x1c
main.main.func2()
        github.com/Jguer/yay/v12/main.go:155 +0x5c
panic({0xaaaabb80ada0, 0x4000020120})
        runtime/panic.go:884 +0x204
github.com/Jguer/yay/v12/pkg/query.(*SourceQueryBuilder).Results(0x400014c0e0, {0xaaaabb876868, 0x40000fc360}, 0x1)
        github.com/Jguer/yay/v12/pkg/query/query_builder.go:215 +0x3f8
main.syncSearch({0xaaaabb872748?, 0x40000a6000?}, {0x4000116d00?, 0x5?, 0x400017bb38?}, {0xaaaabb876868, 0x40000fc360}, {0xaaaabb872908, 0x400014c0e0}, 0x8c?)
        github.com/Jguer/yay/v12/query.go:32 +0xa0
main.handleSync({0xaaaabb872748, 0x40000a6000}, 0x40000b2d80, 0x400012fc20, {0xaaaabb876868, 0x40000fc360})
        github.com/Jguer/yay/v12/cmd.go:354 +0x88c
main.handleCmd({0xaaaabb872748, 0x40000a6000}, 0x40000b2d80, 0x400012fc20, {0xaaaabb876868?, 0x40000fc360?})
        github.com/Jguer/yay/v12/cmd.go:178 +0x2dc
main.main()
        github.com/Jguer/yay/v12/main.go:161 +0x854
```

`yay -Pg`:
```
$ yay -Pg                                                 2023/04/29 12:56:03
{
        "aururl": "https://aur.archlinux.org",
        "aurrpcurl": "https://aur.archlinux.org/rpc?",
        "buildDir": "/home/user/.cache/yay",
        "editor": "",
        "editorflags": "",
        "makepkgbin": "makepkg",
        "makepkgconf": "",
        "pacmanbin": "pacman",
        "pacmanconf": "/etc/pacman.conf",
        "redownload": "no",
        "rebuild": "no",
        "answerclean": "",
        "answerdiff": "",
        "answeredit": "",
        "answerupgrade": "",
        "gitbin": "git",
        "gpgbin": "gpg",
        "gpgflags": "",
        "mflags": "",
        "sortby": "votes",
        "searchby": "name-desc",
        "gitflags": "",
        "removemake": "ask",
        "sudobin": "sudo",
        "sudoflags": "",
        "version": "12.0.4",
        "requestsplitn": 150,
        "completionrefreshtime": 7,
        "maxconcurrentdownloads": 0,
        "bottomup": true,
        "sudoloop": false,
        "timeupdate": false,
        "devel": true,
        "cleanAfter": false,
        "provides": true,
        "pgpfetch": true,
        "upgrademenu": true,
        "cleanmenu": true,
        "diffmenu": true,
        "editmenu": false,
        "combinedupgrade": true,
        "useask": false,
        "batchinstall": true,
        "singlelineresults": false,
        "separatesources": true,
        "newinstallengine": true,
        "debug": false,
        "rpc": true,
        "doubleconfirm": true
}
```

and no changes in `/etc/pacman.conf`.

**Comment by Jguer** [comment]
Hey @jnguk , can you try yay-git? I think that is already solved there. Do you have a repository in your `/etc/pacman.conf` called `aur`?

**Comment by Jguer** [comment]
Closing as dup and fixed #2125

**Comment by segln** [comment]
@Jguer 
Yes I have `aur` repository in `/etc/pacman.conf` and yay-git works fine.

Sorry for confusion.
